### PR TITLE
Format error: add support for 'x' as a formatting option

### PR DESCRIPTION
**Pull request by rupertford**
A Fortran formatting exception allows 'x' to be used without a preceding integer. This PR adds support for this extension. It creates a new EXTENSIONS list which will allow us to control the support of extensions in the future.

### CODE REVIEW

**Comment by rupertford** [comment]
Either @arporter or @TeranIvy could review.

File: src/fparser/two/utils.py [hunk 1]
```diff
@@ -75,6 +75,9 @@
 from fparser.two import pattern_tools as pattern
 from fparser.common.readfortran import FortranReaderBase
 
+# A list of supported extensions to the standard(s)
```
Comment (arporter) [inline src/fparser/two/utils.py]: Please add a brief comment to describe what the "x-format" extension is.

File: src/fparser/two/Fortran2003.py [hunk 2]
```diff
@@ -7283,36 +7282,98 @@ class K(Base):  # R1012
...
+
+        :raises InternalError: if the length of the internal items \
+        list is not 2.
+
+        :raises InternalError: if the second entry of the internal \
+        items list has no content.
+
+        '''
+        if not len(self.items) == 2:
+            raise InternalError(
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py]: This line is not covered.

File: src/fparser/two/Fortran2003.py [hunk 3]
```diff
@@ -7283,36 +7282,98 @@ class K(Base):  # R1012
...
+                "Class Position_Edit_Desc method tostr() has '{0}' items, "
+                "but expecting 2.".format(len(self.items)))
+        if not self.items[1]:
+            raise InternalError(
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py]: This line is not covered.

File: doc/developers_guide.rst [hunk 4]
```diff
@@ -609,14 +609,33 @@ then there would be a one-to-one correspondance between the rules and
...
+
+Compilers often support extensions to the Fortran standard. fparser2
+also does this in certain cases. The suggested way to support this in
+fparser2 is to add an appropriate name to the `EXTENSIONS` list in
+`utils.py` and then support this extension in the appropriate class if
+the name is found in the `EXTENSIONS` list. This will allow this list
+to be modified in the future (e.g. a `-std` option could force the
+compiler to throw out any non-standard Fortran).
+
+Currently this approach is only implemented for a formating extensions
```
[the remark this answers is not in the document]
Comment (rupertford) [reply doc/developers_guide.rst]: Wells spoted.

File: src/fparser/two/Fortran2003.py [hunk 5]
```diff
@@ -7283,36 +7282,98 @@ class K(Base):  # R1012
 
 
 class Position_Edit_Desc(Base):  # R1013
-    """
-    <position-edit-desc> = T <n>
-                           | TL <n>
-                           | TR <n>
-                           | <n> X
+    '''
+    Fortran 2003 rule R1013
```
[the remark this answers is not in the document]
Comment (rupertford) [reply src/fparser/two/Fortran2003.py]: I've been changing every rule from the old format to the new one when I update the class. I do this as the new format is how it is written in the specification document.

File: src/fparser/two/Fortran2003.py [hunk 6]
```diff
@@ -7283,36 +7282,98 @@ class K(Base):  # R1012
...
 
     @staticmethod
     def match(string):
-        if len(string) <= 1:
-            return
-        if string[0].upper() == 'T':
-            if string[1].upper() in 'LR':
-                start = string[:2]
-                rest = string[2:].lstrip()
+        '''Check whether the input matches the rule
```
[the remark this answers is not in the document]
Comment (rupertford) [reply src/fparser/two/Fortran2003.py]: That's a nice easy one :-)

File: src/fparser/two/Fortran2003.py [hunk 7]
```diff
@@ -7283,36 +7282,98 @@ class K(Base):  # R1012
...
+
+        param str string: contains the Fortran that we are trying to
```
[the remark this answers is not in the document]
Comment (rupertford) [reply src/fparser/two/Fortran2003.py]: I always have at least one continuation error :-(

File: src/fparser/two/utils.py (hunk 1 above)
Comment (rupertford) [reply src/fparser/two/utils.py]: OK, I've added some words.

File: src/fparser/two/Fortran2003.py (hunk 2 above)
Comment (rupertford) [reply src/fparser/two/Fortran2003.py]: Oh yes, I forgot about this. Sorry.

File: src/fparser/two/Fortran2003.py (hunk 3 above)
Comment (rupertford) [reply src/fparser/two/Fortran2003.py]: And this one. Sorry.

**Comment by rupertford** [comment]
Reviewers comments have been addressed. Ready for 2nd review from @arporter.

File: src/fparser/two/Fortran2003.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (arporter) [reply src/fparser/two/Fortran2003.py]: Oh, OK. In that case I really don't have a leg to stand/complain on :-)

**Pull request merged**